src/transporters.ts

```diff
@@ -57,18 +57,19 @@ export class DefaultTransporter {
    * @return Configured options.
    */
   configure(opts: GaxiosOptions = {}): GaxiosOptions {
-    opts.headers = opts.headers || {};
+    const headers = Object.assign({}, opts.headers || {});
     if (typeof window === 'undefined') {
       // set transporter user agent if not in browser
-      const uaValue: string = opts.headers['User-Agent'];
-      if (!uaValue) {
-        opts.headers['User-Agent'] = DefaultTransporter.USER_AGENT;
+      const uaValue: string = headers['User-Agent'];
+      if (!headers.hasOwnProperty('User-Agent')) {
+        headers['User-Agent'] = DefaultTransporter.USER_AGENT;
+      } else if (!uaValue) {
+        delete headers['User-Agent'];
```

Comment:
Who do we need to delete the property in this case?

Reply:
I think it's cleaner, I'd rather have the `User-Agent` explicitly not set, rather than it set as an empty string, or `null`, or `undefined`.

Reply:
So... what happens when we delete the `User-Agent` header? Will the underlying transporter not send anything, or will it send some default value?

Reply:
Yeah, that's what I figured: gaxios will send this by default (with no `User-Agent` set):

```
User-Agent: node-fetch/1.0 (+https://github.com/bitinn/node-fetch)
```

Is that what we want?